vllm/model_executor/layers/fused_moe/layer.py

```diff
@@ -246,6 +249,51 @@ def forward_tpu(
     forward_native = forward_cuda
 
 
+def determine_expert_map(
+        ep_size: int, ep_rank: int,
+        global_num_experts: int) -> Tuple[int, Optional[torch.Tensor]]:
+    """
+        Calculates how many experts should be assigned to each rank for EP and
+        creates a mapping from global to local expert index. Experts are
+        distributed evenly across ranks. Any remaining are assigned to the
+        last rank.
```

Comment:
Maybe this behavior could be improved by using the more even partitioning strat from https://github.com/vllm-project/vllm/pull/13839

Reply:
Not a bad idea! I think we can try this in a future PR. cc @cakeng